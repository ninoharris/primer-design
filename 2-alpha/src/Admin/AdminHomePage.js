import React, { Component } from 'react'
import { connect } from 'react-redux'

import Header from './Header'
import ExerciseList from './ExerciseList'
import withLoading from '../components/withLoading'
import FilterExercises from './FilterExercises'

class AdminHomePage extends Component {
  render() {
    return (
      <div className="container-fluid">
        <Header title="Viewing exercises" />
        <div className="row">
          <div className="col-12">
            <FilterExercises />
          </div>
          <div className="col-12">
            <ExerciseList />
          </div>
        </div>
      </div>
    )
  }
}

AdminHomePage = withLoading(true, true)(AdminHomePage, true)

export default connect()(AdminHomePage)