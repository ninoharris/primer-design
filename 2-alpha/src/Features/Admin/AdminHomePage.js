import React, { Component } from 'react'
import { connect } from 'react-redux'
import { Link } from 'react-router-dom'

import ExerciseList from './ExerciseList'
import withLoading from '../../components/withLoading'
import FilterExercises from './FilterExercises'

class AdminHomePage extends Component {
  render() {
    return (
      <div className="container-fluid">
        <div className="row">
          <div className="Nav">
            <div className="Logo">
              <h4>Primer design - admin</h4>
            </div>
            <div>
              <Link to="/admin/create"><button className="btn btn-success mr-3">Add new exercise</button></Link>
              <button className="btn btn-info mr-3" onClick={() => { }}>View students</button>
              <button className="btn btn-warning mr-3" onClick={() => { }}>Log out</button>
            </div>
          </div>
        </div>
        <div className="row">
          <div className="col-12">
            <FilterExercises />
          </div>
          <div className="col-12">
            <ExerciseList />
          </div>
        </div>
      </div>
    )
  }
}

AdminHomePage = withLoading(true, true)(AdminHomePage, true)

export default connect()(AdminHomePage)