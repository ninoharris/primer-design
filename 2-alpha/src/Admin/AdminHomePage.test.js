import React from 'react'
import { shallow } from 'enzyme'
import { AdminHomePage } from './AdminHomePage'

test('AdminHomePage shallow renders correctly', () => {
  const wrapper = shallow(<AdminHomePage />)
  expect(wrapper).toMatchSnapshot()
})