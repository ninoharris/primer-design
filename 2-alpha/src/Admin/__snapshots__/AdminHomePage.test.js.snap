// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminHomePage shallow renders correctly 1`] = `
<div
  className="container-fluid"
>
  <withRouter(Header)
    title="Viewing exercises"
  />
  <div
    className="row"
  >
    <div
      className="col-12"
    >
      <Connect(ExerciseListFilters) />
    </div>
    <div
      className="col-12"
    >
      <Connect(ExerciseList) />
    </div>
  </div>
</div>
`;
